tests/pplat.cu

```diff
@@ -53,21 +53,46 @@ __global__ void pp_kernel(uint32_t *d_buf, uint32_t *h_buf, uint32_t num_iters)
     }
 }
 
-int dev_id = 0;
-uint32_t num_iters = 1000;
+static int dev_id = 0;
+static uint32_t num_iters = 1000;
+static unsigned int timeout = 0;  // in s
 
-void print_usage(const char *path)
+static void print_usage(const char *path)
 {
-    cout << "Usage: " << path << " [-h][-d <gpu>][-t <iters>][-a <fn>]" << endl;
+    cout << "Usage: " << path << " [-h][-d <gpu>][-t <iters>][-u <timeout>][-a <fn>]" << endl;
     cout << endl;
     cout << "Options:" << endl;
     cout << "   -h              Print this help text" << endl;
     cout << "   -d <gpu>        GPU ID (default: " << dev_id << ")" << endl;
     cout << "   -t <iters>      Number of iterations (default: " << num_iters << ")" << endl;
+    cout << "   -u <timeout>    Timeout in second. 0 to disable. (default: " << timeout << ")" << endl;
     cout << "   -a <fn>         GPU buffer allocation function (default: cuMemAlloc)" << endl;
     cout << "                       Choices: cuMemAlloc, cuMemCreate" << endl;
 }
 
+/**
+ * Return time difference in us.
+ */
+static inline double time_diff(struct timespec start, struct timespec end)
+{
+    return (double)((end.tv_nsec - start.tv_nsec) / 1000.0 + (end.tv_sec - start.tv_sec) * 1000000.0);
+}
+
+static inline void check_timeout(struct timespec start, double timeout_us)
+{
+    struct timespec now;
+    double time_used_us;
+    if (timeout_us > 0) {
+        clock_gettime(MYCLOCK, &now);
+        time_used_us = time_diff(start, now);
+        if (time_used_us > timeout_us) {
+            cerr << "ERROR: TIMEOUT!!!" << endl;
+            ASSERTDRV(cuStreamQuery(0));
```

Comment:
- CUDA_ERROR_NOT_READY would mean that the kernel is  still running, so that would be a timeout caused by a hanging kernel
- CUDA_SUCCESS would mean that the kernel has exited, which is still unexpected, so it is not a good exit value either
- any other error would be unexpected.
I would just print and decode any code returned here

Reply:
Changed the code to report status returned from `cuStreamQuery(0)`.